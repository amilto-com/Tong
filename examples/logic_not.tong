// Demonstrate unary logical negation '!'
// Expected output:
// NOT1 true
// NOT2 false
// Combined true

fn main() {
  let a = false
  let b = !a
  print("NOT1", b)
  let c = !b
  print("NOT2", c)
  let d = !(a || b) & (!c || a == false)
  print("Combined", d)
}

main()
