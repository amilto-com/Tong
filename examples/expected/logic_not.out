NOT1 true
NOT2 false
Combined false
